app/assets/javascripts/modules/project.js

```diff
@@ -70,4 +70,92 @@ angular.module('probedock.project', [ 'probedock.api', 'probedock.forms', 'probe
     }
   })
 
-;
+.directive('projectHealthChart', function() {
+  return {
+    restrict: 'E',
+    controller: 'ProjectHealthChartCtrl',
+    templateUrl: '/templates/project-health.html',
+    scope: {
+      project: '='
+    }
+  };
+})
+
+
+.controller('ProjectHealthChartCtrl', function(api, $scope) {
+  $scope.chart = {
+    data: [],
+    labels: ['passed', 'failed', 'inactive'],
+    colors: ['#62c462', '#ee5f5b', '#fbb450'],
+    params: {}
+
+  };
+
+  $scope.projectVersionChoices = [];
+
+  $scope.$watch('project', function(value) {
+    if (value) {
+      fetchMetrics();
+      fetchProjectVersionChoices();
+    }
+  });
+
+  var ignoreChartParams = true;
+  $scope.$watch('chart.params', function(value) {
+    if (value && !ignoreChartParams) {
+      fetchMetrics();
+    }
+
+    ignoreChartParams = false;
+  }, true);
+
+  function fetchProjectVersionChoices() {
+    api({
+      url: '/projectVersions',
+      params: {
+        projectId: $scope.project.id
+      }
+    }).then(function(res) {
+      $scope.projectVersionChoices = res.data;
+    });
+  }
+
+  function fetchMetrics() {
+    var params;
```

Comment:
You could remove 4 lines from the if/else below by writing `params = {};` here and simply adding the properties to the object. It would also be easier to add base properties in the future.